warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/handler/edge/ProtobufEdgeDataTypeHandler.java

```diff
@@ -741,6 +609,183 @@ public long process(KEYIN key, RawRecordContainer event, Multimap<String,Normali
         return edgesCreated;
     }
     
+    private boolean shortCircuit(RawRecordContainer event, String typeName) {
+        if (event.fatalError()) {
+            return true;
+        }
+        if (!edges.containsKey(typeName)) {
+            return true;
+        }
+        return false;
+    }
+    
+    private boolean eventLacksEdgeFields(EdgeDefinition edgeDef) {
+        
+        Multimap<String,NormalizedContentInterface> mSource = null;
+        Multimap<String,NormalizedContentInterface> mSink = null;
+        
+        if (depthFirstList.containsKey(edgeDef.getSourceFieldName()) && depthFirstList.containsKey(edgeDef.getSinkFieldName())) {
+            mSource = depthFirstList.get(edgeDef.getSourceFieldName());
+            mSink = depthFirstList.get(edgeDef.getSinkFieldName());
+        } else {
+            return true;
+        }
+        
+        // bail if the event doesn't contain any values for the source or sink field
+        if (null == mSource || null == mSink) {
+            return true;
+        }
+        if (mSource.isEmpty() || mSink.isEmpty()) {
+            return true;
+        }
+        
+        return false;
+    }
+    
+    private long getActivityDate() {
+        if (normalizedFields.containsKey(edgeDefConfigs.getActivityDateField())) {
```

Comment:
I like that this method was extracted.  I'm concerned that the parsing may be redone though for each edge that needs it.  It might be a good idea to store the activity date as a field to avoid repeated parsing.